modules/configuration-api/src/main/java/org/apache/ignite/configuration/ConfigurationModule.java

```diff
@@ -136,4 +138,24 @@ default void patchConfigurationWithDynamicDefaults(SuperRootChange rootChange) {
     default void migrateDeprecatedConfigurations(SuperRootChange superRootChange) {
         // No-op.
     }
+
+    /**
+     * Returns a collection of prefixes, removed from configuration. Keys that match any of the prefixes
+     * in this collection will be deleted.
+     * <p>Use ignite.my.deleted.property for regular deleted properties</p>
+     * <p>ignite.list.*.deletedProperty - for named list elements. Arbitrarily nested named lists are supported</p>
+     *
+     * @return A collection of prefixes of deleted keys.
+     */
+    default Collection<String> deletedPrefixes() {
+        return emptySet();
+    }
+
+    /** Compiles @{link #deletedPrefixes()} into regex patterns. Shouldn't be overwritten. */
+    default Collection<Pattern> deletedPrefixesPatterns() {
```

Comment:
Can it be a static method? I don't trust this `Shouldn't be overwritten` in interface